### PR TITLE
Try to delete nodes in all cloud zones

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
Testnet automation is failing to cleanup nodes after completing the runs.

#### Summary of Changes
The testnet is being deployed in multiple regions. The cleanup script tries to delete all nodes assuming a single region. Changed the script to delete nodes in all zones.

### CODE REVIEW

File: net/scripts/gce-provider.sh [hunk 1]
```diff
@@ -208,10 +208,11 @@ cloud_DeleteInstances() {
   declare names=("${instances[@]/:*/}")
   declare zones=("${instances[@]/*:/}")
 
-  (
+  for zone in "${zones[@]}"; do
     set -x
-    gcloud beta compute instances delete --zone "${zones[0]}" --quiet "${names[@]}"
-  )
+    # Try deleting instances in all zones
+    gcloud beta compute instances delete --zone "$zone" --quiet "${names[@]}" || true
```
Comment (danpaul000) [inline net/scripts/gce-provider.sh]: What is the purpose of the "|| true"?

Comment (pgarg66) [reply net/scripts/gce-provider.sh]: Ignore the error of the command. If it fails to delete some nodes in one zone, it'll continue with deleting in other zones.

**Comment by pgarg66** [comment]
> Are you looking at this as a solution to #3821 ?

Yes, didn't realize there was an issue for it.

**Comment by pgarg66** [comment]
Fixes: https://github.com/solana-labs/solana/issues/3821

File: net/scripts/gce-provider.sh (hunk 1 above)
Comment (danpaul000) [reply net/scripts/gce-provider.sh]: Do we want to act on that failure, or just suppress and move on?  One teardown failure should not kill the script, but does that leave us with a dangling node?

**Comment by danpaul000** [comment]
> > Are you looking at this as a solution to #3821 ?
> 
> Yes, didn't realize there was an issue for it.

I'm trying to be diligent about keeping an up to date queue of all issues or broken things in the dev-ops realm as they come up.  Perhaps I just over-zealously self-assigned. :)

**Comment by pgarg66** [comment]
> > > Are you looking at this as a solution to #3821 ?
> > 
> > 
> > Yes, didn't realize there was an issue for it.
> 
> I'm trying to be diligent about keeping an up to date queue of all issues or broken things in the dev-ops realm as they come up. Perhaps I just over-zealously self-assigned. :)

It's great to keep a queue of outstanding issues. Typically we've been fixing stuff as we see it broken. Sorry, didn't realize you were already working on it.

File: net/scripts/gce-provider.sh (hunk 1 above)
Comment (pgarg66) [reply net/scripts/gce-provider.sh]: I think failures in deletion in one zone should not prevent the script from deleting nodes in other zones. Some sort of audit towards the end and reporting will be useful (i.e. if some node is still lingering around, notify the user).

**Comment by danpaul000** [comment]
> > > > Are you looking at this as a solution to #3821 ?
> > > 
> > > 
> > > Yes, didn't realize there was an issue for it.
> > 
> > 
> > I'm trying to be diligent about keeping an up to date queue of all issues or broken things in the dev-ops realm as they come up. Perhaps I just over-zealously self-assigned. :)
> 
> It's great to keep a queue of outstanding issues. Typically we've been fixing stuff as we see it broken. Sorry, didn't realize you were already working on it.

No worries.  I hadn't started work on that issue yet anyway.  I'm not quite fast enough yet to fix things as they are appearing while working on the Azure bring-up in parallel, hence the queue.  Let's just continue to be in sync for these kinds of issues so we know who will take them up.

**Pull request merged**